vic/drivers/shared_image/src/check_domain_info.c

```diff
@@ -0,0 +1,81 @@
+/******************************************************************************
+ * @section DESCRIPTION
+ *
+ * Routines to compare the global domain to the other VIC input files
+ *
+ * @section LICENSE
+ *
+ * The Variable Infiltration Capacity (VIC) macroscale hydrological model
+ * Copyright (C) 2016 The Computational Hydrology Group, Department of Civil
+ * and Environmental Engineering, University of Washington.
+ *
+ * The VIC model is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *****************************************************************************/
+
+ #include <vic_driver_shared_image.h>
+
+/******************************************************************************
+ * @brief    Check that the cooridnates, dimensions, and mask variables in a
+             netcdf file matches the global domain.
+ *****************************************************************************/
+void
+compare_ncdomain_with_global_domain(char *ncfile)
+{
+    extern domain_struct global_domain;
+
+    domain_struct        ncfile_domain;
+
+    size_t               i;
+
+    ncfile_domain.info = global_domain.info;
+
+    get_global_domain(ncfile, &ncfile_domain, true);
+
+    // checks
+
+    // dimension shapes match (lat/lon)
+    if (global_domain.n_nx != ncfile_domain.n_nx) {
+        log_err("x dimension in parameters file does not match domain");
+    }
+    if (global_domain.n_ny != ncfile_domain.n_ny) {
+        log_err("y dimension in parameters file does not match domain");
+    }
+
+    for (i = 0; i < global_domain.ncells_total; i++) {
```

Comment:
This section loops through every cell and check that each matches? Maybe make that distinction with a comment?